libs/core/execution_base/include/hpx/execution_base/any_sender.hpp

```diff
@@ -0,0 +1,746 @@
+        {
+            if (&other != this)
+            {
+                if (!empty())
+                {
+                    release();
+                }
+
+                if (!other.empty())
+                {
+                    if (other.using_embedded_storage())
+                    {
+                        std::memcpy(embedded_storage, other.embedded_storage,
+                            embedded_storage_size);
+                        object = reinterpret_cast<base_type*>(embedded_storage);
+                    }
+                    else
+                    {
+                        heap_storage = other.heap_storage;
+                        other.heap_storage = nullptr;
+                        object = heap_storage;
+                    }
+
+                    is_empty = false;
+
+                    other.object = get_empty_vtable<base_type>::call();
+                    other.is_empty = true;
+                }
+            }
+
+            return *this;
+        }
+
+        base_type const& get() const noexcept
+        {
+            return *object;
+        }
+
+        base_type& get() noexcept
+        {
+            return *object;
+        }
+
+        template <typename Impl, typename... Ts>
+        void store(Ts&&... ts)
+        {
+            if (!empty())
+            {
+                release();
+            }
+
+            if constexpr (can_use_embedded_storage<Impl>())
+            {
+                new (embedded_storage) Impl(std::forward<Ts>(ts)...);
+                object = reinterpret_cast<base_type*>(embedded_storage);
+            }
+            else
+            {
+                heap_storage = new Impl(std::forward<Ts>(ts)...);
+                object = heap_storage;
+            }
+
+            is_empty = false;
+        }
+    };
+
+    template <typename Base, std::size_t EmbeddedStorageSize,
+        std::size_t AlignmentSize = sizeof(void*)>
+    class copyable_sbo_storage
+      : public sbo_storage<Base, EmbeddedStorageSize, AlignmentSize>
+    {
+        using movable_sbo_storage =
+            sbo_storage<Base, EmbeddedStorageSize, AlignmentSize>;
+
+        using typename movable_sbo_storage::base_type;
+
+        using movable_sbo_storage::embedded_storage;
+        using movable_sbo_storage::empty;
+        using movable_sbo_storage::heap_storage;
+        using movable_sbo_storage::is_empty;
+        using movable_sbo_storage::object;
+        using movable_sbo_storage::release;
+        using movable_sbo_storage::using_embedded_storage;
+
+    public:
+        using movable_sbo_storage::get;
+        using movable_sbo_storage::store;
+
+        copyable_sbo_storage() = default;
+        copyable_sbo_storage(copyable_sbo_storage&&) = default;
+        copyable_sbo_storage& operator=(copyable_sbo_storage&&) = default;
+
+        copyable_sbo_storage(copyable_sbo_storage const& other)
+          : movable_sbo_storage()
+        {
+            *this = other;
+        }
```

Comment:
```suggestion
        copyable_sbo_storage(copyable_sbo_storage const& other)
          : movable_sbo_storage(other)
        {
        }
```

Reply:
Those are not the same though. `sbo_storage` (the alias `movable_sbo_storage`) does not have a copy-constructor.

Reply:
Urgs.

Reply:
How should I interpret this comment? ;)

Reply:
I simply find it to be unusual that a class has a `foo& operator=(foo const&)` without having a `foo(foo const&)`

Reply:
But `copyable_sbo_storage` has both the copy constructor and copy assignment operator defined... what am I missing? What did I mess up? ;) And `movable_sbo_storage` has neither (though I haven't explicitly deleted them; will do so now).

Reply:
Right, I misspoke. I guess what bothered me then was that you implemented the copy constructor in terms of the assignment operator. `*this = std::move(other)` first destructs `*this` and then constructs a copy of the rhs.

Reply:
I see, that makes more sense. Thanks. I'll have another look at this.

Reply:
I actually still don't see it, and this might be due to gaps in my understanding of constructors and assignment operators. There's at least one mixup here. You're `move`-ing `other`, but talking about the copy constructor, so I assume you meant `*this = other`, correct?

At what point is `*this = other` going to do destruct `*this`? Is it because `copyable_sbo_storage& operator=(copyable_sbo_storage const& other)` is not ref qualified? Can or should the copy assignment operator generally actually be ref qualified? In practice it looks like the extra constructor and destructor calls are elided, but I guess you're saying this is not guaranteed?kkk

Whatever the case is, I've moved the copy and move assignment into helper functions and call them from the copy/move constructors/assignment operators instead.